### ISSUE TITLE
add_regs param of make_design_matrix() in design_matrix.py ishould handle the format of the passed in data transparently.

### ISSUE THREAD

**Issue by kchawla-pi** [nistats/nistats#198]
(Issue may require more fleshing out.)

In design_matrix.py > make_design_matrix(add_regs), the add_regs parameter is very specific. Essentially, everywhere the events are passed in, the API should handle the formats transparently.
Currently, the two formats are:
 1. vector with same number of elements as vol in scans.
 2. onset duration & amplitude which is converted to a dense representation.
Which ones are passed in depends on the API.
This should not be the case. The program should determine the format passed in and deal appropriately automatically.

**Comment by bthirion** [comment]
I agree that the library sticks to old conventions here. I'm not sure howto implement that though. Suggestions welcome !

**Comment by bthirion** [comment]
- See also nilearn/nilearn#2377

**Comment by kchawla-pi** [comment]
"We need to be explicit, maybe improve the API for being explicit"?
